packages/react-core/src/components/Button/Button.tsx

```diff
@@ -58,6 +64,8 @@ export interface ButtonProps extends Omit<React.HTMLProps<HTMLButtonElement>, 'r
   inoperableEvents?: string[];
   /** Adds inline styling to a link button */
   isInline?: boolean;
+  /** Adds small or large styling to the button */
+  size?: 'sm' | 'default' | 'lg';
```

Comment:
@mcoker Yes, I can see where we have lots of inconsistency between how we name these props across the library and it would be good to have a system that can be applied everywhere. Do you have a proposal for what that should be? It does seem like for some elements, like buttons, it's strictly about size. For other elements it also involves different spacing.  cc @mceledonia

Reply:
@mcoker this [issue](https://github.com/patternfly/patternfly-react/issues/8098) is also open and relevant to what you are saying.   From the issue 

> Describe the problem
> Accordion, DescriptionList, Expandable Section all have display size props and the are not define consistently.
> 
> Accordion and Expandable section have:
> displaySize?: 'default' | 'large';
> 
> DescriptionList has
> displaySize?: 'lg' | '2xl';
> 
> I propose we change them to follow closely to how we name other size variants
> 
> Accordion and Expandable section have:
> displaySize?: 'default' | 'lg';
> and
> DescriptionList:
> displaySize?: 'default' | 'lg' | '2xl';

Reply:
@tlabaj nice! I didn't see that one, and I overlooked expandable section above. Thank you!

> Do you have a proposal for what that should be? It does seem like for some elements, like buttons, it's strictly about size. For other elements it also involves different spacing.

@mcarrano I don't... naming things is so hard! And you're right - we have size variations that (this is just a sample)
* Just change the box dimensions - modals
* change the overall size of the thing, including font, spacing, box size if it's an inline element - empty state, tables, progress, spinner, button, icon, title, compact labels/cards/tables... probably most things
* change the gutter spacing (description list)
* change the presentation - compact pagination, compact tree view, maybe some of the display-lg variations?

Most of these can't be used with any other kind of size variation. Some can though
* description list's compact variation changes gutter size, and we could offer another variation that changes the font size, though there's probably an argument to keep those together. Or just change our current compact variation to something like pf-m-spacer-sm or gutter-sm or gap-sm, kind of like how we have inset (left/right edge spacing) using pf-m-inset-[size]
* modal's size variations just change the outer box size. You may want to combine that with a future modifier that changes the font size or spacing or something like that.

I think we should move on with this PR though, so @gitdallas can move on with his life 😂 I created a core issue if it's something we want to look at - https://github.com/patternfly/patternfly/issues/5155

Reply:
That sounds good to me!